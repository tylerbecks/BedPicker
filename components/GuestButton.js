import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

const GuestButton = ({ guestName, onPress, selected }) => (
  <TouchableOpacity
    style={
      selected ? { ...styles.button, ...styles.selectedButton } : styles.button
    }
    onPress={() => {
      onPress(guestName);
    }}
  >
    <Text
      style={
        selected ? { ...styles.text, ...styles.selectedText } : styles.text
      }
    >
      {guestName}
    </Text>
  </TouchableOpacity>
);

export default GuestButton;

const BUTTON_SIZE = 90;

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: '#CFD8DC',
    borderRadius: BUTTON_SIZE,
    height: BUTTON_SIZE,
    justifyContent: 'center',
    padding: 10,
    width: BUTTON_SIZE,
    margin: 10
  },
  selectedButton: {
    backgroundColor: 'blue'
  },
  selectedText: {
    color: 'white'
  },
  text: {
    color: 'black',
    fontWeight: 'bold'
  }
});
