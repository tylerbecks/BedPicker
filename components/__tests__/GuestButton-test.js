import 'react-native';
import React from 'react';
import GuestButton from '../GuestButton';
import renderer from 'react-test-renderer';

it('renders with blue background when selected', () => {
  expect(
    renderer
      .create(<GuestButton guestName="foo" onPress={jest.fn()} selected />)
      .toJSON()
  ).toMatchSnapshot();
});

it('renders with gray background when not selected', () => {
  expect(
    renderer
      .create(
        <GuestButton guestName="foo" onPress={jest.fn()} selected={false} />
      )
      .toJSON()
  ).toMatchSnapshot();
});
