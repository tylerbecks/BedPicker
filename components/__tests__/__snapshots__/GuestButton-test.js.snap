// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with blue background when selected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 90,
      "height": 90,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
      "width": 90,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    foo
  </Text>
</View>
`;

exports[`renders with gray background when not selected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CFD8DC",
      "borderRadius": 90,
      "height": 90,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
      "width": 90,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontWeight": "bold",
      }
    }
  >
    foo
  </Text>
</View>
`;
